core/templates/components/state-directives/rule-editor/rule-editor.directive.ts

```diff
@@ -258,10 +258,6 @@ angular.module('oppia').directive('ruleEditor', [
           };
 
           ctrl.$onInit = function() {
-            $scope.$on('updateAnswerGroupInteractionId', function(
```

Comment:
Should there be a replacement for this?

Reply:
So, there is only one broadcast corresponding to this, and it doesn't pass any value. I assume there had been broadcasts earlier which had been triggering this which have since been removed. I tested this code and confirmed with @bansalnitish that we do not, in fact need this, since now to change the Interaction Id, one has to delete the interaction and then select a new one through a new modal

Reply:
Hm, can you please test this scenario:
1. Create an exploration with multiple states (at least 4).
2. Have different interactions in each of these states with each having multiple rules and AnswerGroups.
3. Switch between the states in the editor and check if the Rules and AnswerGroups show expected values.

Reply:
@kevintab95 Yes, it works!